doc/source/whatsnew/v0.18.1.txt

```diff
@@ -138,6 +138,47 @@ API changes
 - Provide a proper ``__name__`` and ``__qualname__`` attributes for generic functions (:issue:`12021`)
 - ``pd.concat(ignore_index=True)`` now uses ``RangeIndex`` as default (:issue:`12695`)
 
+.. _whatsnew_0181.enhancements.groubynth:
+
+Index in ``Groupby.nth`` output is now more consistent with ``as_index``
+argument passed in (:issue:`11039`):
+
+Previous Behavior:
+
+.. code-block:: ipython
+
+    In [4]: df
+    Out[4]:
+       A  B
+    0  a  1
+    1  b  2
+    2  a  3
+
+    In [5]: df.groupby('A', as_index=True)['B'].nth(0)
+    Out[5]:
+    0    1
+    1    2
+    Name: B, dtype: int64
```

Comment:
the point here is that `as_index=True` is ignored in old behaviour; `as_index=False` is not relevant or informative (and has not changed).

this has been going back and forth for too many times. if you like to add/modify anything please do so on your end.